superset-frontend/src/components/Chart/DrillBy/DrillByModal.tsx

```diff
@@ -0,0 +1,101 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import {
+  BinaryQueryObjectFilterClause,
+  QueryFormData,
+  t,
+  useTheme,
+} from '@superset-ui/core';
+import Modal from 'src/components/Modal';
+import Button from 'src/components/Button';
+import { useSelector } from 'react-redux';
+import { Slice } from 'src/types/Chart';
+
+interface ModalFooterProps {
+  exploreChart: () => void;
+  closeModal?: () => void;
+}
+
+const ModalFooter = ({ exploreChart, closeModal }: ModalFooterProps) => (
+  <>
+    <Button buttonStyle="secondary" buttonSize="small" onClick={exploreChart}>
+      {t('Edit chart')}
+    </Button>
+    <Button
+      buttonStyle="primary"
+      buttonSize="small"
+      onClick={closeModal}
+      data-test="close-drillby-modal"
+    >
+      {t('Close')}
+    </Button>
+  </>
+);
+
+interface DrillByModalProps {
+  chartId: number;
+  formData: QueryFormData;
+  initialFilters: BinaryQueryObjectFilterClause[];
+  showModal: boolean;
+  onHideModal: () => void;
+}
+
+export default function DrillByModal({
+  chartId,
+  formData,
+  initialFilters,
+  showModal,
+  onHideModal,
+}: DrillByModalProps) {
+  console.log(formData);
+  console.log(initialFilters);
+  const theme = useTheme();
+  const { slice_name: chartName } = useSelector(
```

Comment:
User can override slice name in dashboard edit mode. Could we use that? Like `state.dashboardLayout.present[chartLayoutItemId].meta.sliceNameOverride || state.dashboardLayout.present[chartLayoutItemId].meta.sliceName`

Reply:
is the drill by available in dashboard edit mode?

Reply:
It's not, but user can change chart's displayed name in edit mode. For example, you add chart "My chart" to a dashboard, then in edit mode you rename that chart to "This is my chart" and save. In such case we'd want "This is my chart" to be displayed in drill by modal